### PR TITLE
HAWQ-1084. Fixed memory allocation for table printing.

### PR DESCRIPTION
**Pull request by sansanichfb**

### CODE REVIEW

File: src/bin/psql/describe.c [hunk 1]
```diff
@@ -4316,10 +4316,10 @@ describePxfTable(const char *profile, const char *pattern, bool verbose)
 		if (verbose)
 		{
 			sourcefieldtype = PQgetvalue(res, i, 4);
-			total_fields = PQgetvalue(res, i, 5);
+			total_fields = atoi(PQgetvalue(res, i, 5));
```
Comment (kavinderd) [inline src/bin/psql/describe.c]: Why was an OOM being thrown without `atoi`?

Comment (kavinderd) [inline src/bin/psql/describe.c]: Ah, good catch

**Comment by kavinderd** [comment]
+1

File: src/bin/psql/describe.c (hunk 1 above)
Comment (GodenYao) [inline src/bin/psql/describe.c]: does atoi() guarantee a positive value?  If there's a chance the char* can be interpreted as negative integer - should we use unsigned int as total_fields is for memory allocation?

**Pull request closed without merging**